### PR TITLE
Adding PSversion to userAgent for every powershell cmdlet

### PR DESCRIPTION
**Pull request by shahabhijeet**
Adding PsVersion to userAgent
Adding test for userAgent
Fixing file misspell

### CODE REVIEW

File: src/Common/Commands.Common/AzurePSCmdlet.cs [hunk 1]
```diff
@@ -243,9 +267,10 @@ protected virtual void TearDownDebuggingTraces()
 
         protected virtual void SetupHttpClientPipeline()
         {
-            ProductInfoHeaderValue userAgentValue = new ProductInfoHeaderValue(
-                ModuleName, string.Format("v{0}", ModuleVersion));
-            AzureSession.ClientFactory.UserAgents.Add(userAgentValue);
+            //We do not want to add a prefix 'v' to version, because when data will be read out of the telemetry, it will again have to be parsed to extract the actualy version
+            AzureSession.ClientFactory.UserAgents.Add(new ProductInfoHeaderValue(ModuleName, ModuleVersion));
```
Comment (markcowl) [inline src/Common/Commands.Common/AzurePSCmdlet.cs]: This is removing the 'v' from the version value

File: src/Common/Commands.Common/AzurePSCmdlet.cs [hunk 2]
```diff
@@ -58,6 +60,28 @@ protected virtual bool IsErrorMetricEnabled
...
+        /// Indicates installed PowerShell version
+        /// </summary>
+        private string _psVersion;
+
+        /// <summary>
+        /// Get PsVersion returned from PowerShell.Runspace instance
+        /// </summary>
+        protected string PSVersion
+        {
+            get
```
Comment (markcowl) [inline src/Common/Commands.Common/AzurePSCmdlet.cs]: This value is available from Cmdlet.CommandRuntime.Host.Version.  You will need to make sure that Host is not null.

**Comment by shahabhijeet** [comment]
Address issue
https://github.com/Azure/azure-powershell/issues/3026

File: src/Common/Commands.Common/AzurePSCmdlet.cs [hunk 3]
```diff
@@ -73,8 +73,15 @@ protected string PSVersion
...
                 {   
-                    System.Management.Automation.PowerShell ps = System.Management.Automation.PowerShell.Create();
-                    _psVersion = ps.Runspace.Version.ToString();
+                    if(this.Host != null)
+                    {
+                        _psVersion = this.Host.Version.ToString();
+                    }
+                    else
+                    {
+                        System.Management.Automation.PowerShell ps = System.Management.Automation.PowerShell.Create();
```
Comment (markcowl) [inline src/Common/Commands.Common/AzurePSCmdlet.cs]: I would say we can return a default value in this case.  This should only occur in tests

**Comment by shahabhijeet** [comment]
PS OnDemand
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1271/

**Comment by markcowl** [comment]
closing in favor of the new PR

**Pull request closed without merging**